packages/neon-api/src/txService/builder.ts

```diff
@@ -0,0 +1,186 @@
+import {
+  Transaction,
+  TransactionAttributeLike,
+  WitnessLike,
+  TransactionAttribute,
+  Witness,
+  CosignerLike
+} from "@cityofzion/neon-core/lib/tx";
+import { ScriptIntent, createScript } from "@cityofzion/neon-core/lib/sc";
+import { reverseHex, Fixed8 } from "@cityofzion/neon-core/lib/u";
+import { Account } from "@cityofzion/neon-core/lib/wallet";
+
+export interface CosignerWithAccount
+  extends Pick<CosignerLike, Exclude<keyof CosignerLike, "account">> {
+  account: Account;
+}
+
+export interface TransactionBuilderConfig {
+  sender: Account | string;
+  intents?: ScriptIntent[];
+  cosigners?: (CosignerWithAccount | CosignerLike)[];
+  systemFee?: number;
+  networkFee?: number;
+  validUntilBlock?: number;
+  attributes?: TransactionAttributeLike[];
+  scripts?: WitnessLike[];
+  script?: string;
+}
+
+/**
+ * `TransactionBuilder` is to build a transaction in a more convinient and mature way.
+ * It is more powerf
+ */
+export class TransactionBuilder {
+  private _sender: Account;
+  /**
+   * Cosigners
+   */
+  private _cosigners: CosignerLike[] = [];
+  /**
+   * Cosigner accounts that client has access to
+   */
+  private _cosignerAccounts: Account[] = [];
+  private _intents: ScriptIntent[] = [];
+  private _transaction: Transaction;
```

Comment:
Having the transaction field and also the other fields might be confusing. Consider either having only the transaction field or just the intermediate fields.

Reply:
I use the builder style and removed transaction.